### DISCUSSION TITLE
Aufsetzen eines neuen React Projects - welche Files kann ich löschen und welche müssen behalten werden?

### DISCUSSION THREAD

**Discussion by TobiasSteinhagen** [neuefische/questions#724] [Web: Self-directed learning] [answered]
Moin,

ich habe eine neue React App aufgesetzt und sitze jetzt vor der Struktur und frage mich, welche Files/Folder ich wirklich davon brauche und ob es ggf. Files und/oder Folder gibt, die ich einfach löschen kann.

![Bildschirm­foto 2023-09-07 um 14 10 32](https://github.com/neuefische/questions/assets/140393240/a2b980aa-3294-4437-a320-c599aa251a67)

Freue mich auf ein Feedback!

VG,
Tobi

**Comment by JessicaLoers** [comment, marked as answer]
**Hi Tobias ✌️** 
diese Frage ist mal wieder mit  _"es kommt darauf an"_ zu beantworten. 

Hier einige Beispiele, bei denen du dich entscheiden kannst: 

`public/favicon.ico`
Dies ist das Standard-Favicon für die Anwendung. Du kannste es durch ein eigenes ersetzen oder anpassen, wenn du möchtest.

`public/logoXXX.png`
Das ist das Logo, das im  Standard-Create-React-App Template angezeigt wird.
Wenn du diese Dateien löschst,  wird das Projekt weiterhin funktionieren, **solange  du alle Verweise auf diese Datei aus Ihrem Quellcode entfernst! ❗️** 

`public/robots.txt`
Das Löschen kann Auswirkungen auf die Art und Weise haben, wie Suchmaschinen die  Website durchsuchen und indizieren. 

`public/manifest.json`
Diese Datei enthält Metadaten und Konfigurationseinstellungen für Progressiv web App (PWA). Theoretisch könntest du diese löschen, da sie  normalerweise keine spürbaren Auswirkungen auf die Funktionsweise der Web-Anwendung hat, aber. Ich würde sie behalten. 


`src/app.test.js`
Das ist eine Testdatei für die React App. Wenn du keinen Test schreibst, kannst du sie löschen. Falls du später Tests schreiben möchten, kannst du diese Datei neu erstellen oder eine neue Testdatei erstellen. 

`src/setup.test.js`
Wird für Testkonfigurationen und -erweiterungen verwendet. Wenn du keine Tests schreibst oder du keine Konfigurationen oder -Erweiterungen für die Tests hinzufügen möchten benötigst du diese Datei auch nicht. 


`src/logo.svg`
ist das react logo und kann gehen, **solange  du alle Verweise auf diese Datei aus Ihrem Quellcode entfernst!❗️** 

`src/reportWebVitals.js`
Das Löschen hat normalerweise keine Auswirkungen auf die tatsächliche Funktionalität der App,  **solange  du alle Verweise auf diese Datei aus Ihrem Quellcode entfernst!❗️**  Das Thema Web Vitals ist etwas umfangreicher hier ist ein auf die schnelle gefundener [Artikel](https://web.dev/vitals/) dazu.
